packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/promo/promo.component.html

```diff
@@ -0,0 +1,49 @@
+<ng-container [ngTemplateOutlet]="variant"></ng-container>
+
+<ng-template #default>
+  <div class="component promo {{ styles }}" [attr.id]="id">
+    <div class="component-content">
```

Comment:
This can be shortened for ease of reading with *ngIf and such.
Top two divs are the same between three variants. Several next divs are the same between default and with text

Reply:
SXA team says withText variant is no longer used in xmcloud and can be removed. With it gone, the #empty template can be activated via inline *ngIf on the html file, to avoid duplicate markup.